zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -558,74 +577,107 @@ impl<F: Field> SignVerifyChip<F> {
 
-                Ok(assigned_ecdsas)
+                let advice_rows = ctx.advice_rows["ecdsa chip"].iter();
+                println!(
+                    "maximum rows used by an advice column: {}",
+                    advice_rows.clone().max().or(Some(&0)).unwrap(),
+                );
+                println!(
+                    "row counts: {:?}",
+                    advice_rows,
+                );
+
+                Ok((assigned_ecdsas, sign_data_decomposed_vec))
             },
         )?;
+
         let (deferred_keccak_check, assigned_sig_verifs) = layouter.assign_region(
             || "signature address verify",
             |region| {
-                let mut assigned_sig_verifs = Vec::new();
-
-                let mut deferred_keccak_check = Vec::new();
                 let mut ctx = Context::new(
                     region,
                     ContextParams {
-                        num_advice: vec![("ecdsa chip".to_string(), NUM_ADVICE)],
+                        num_advice: vec![("ecdsa chip".to_string(),  NUM_ADVICE + 1)],
                     },
                 );
+
+                // IMPORTANT: Move to Phase2 before RLC
+                ctx.next_phase();
+
+                // ================================================
+                // step 3: compute RLC of keys and messages
+                // ================================================
+                let mut assigned_sig_verifs = Vec::new();
+                let mut deferred_keccak_check = Vec::new();
                 for (i, e) in assigned_ecdsas.iter().enumerate() {
                     let sign_data = signatures.get(i); // None when padding (enabled when address == 0)
+                    let sign_data_decomposed = &sign_data_decomposed_vec[i];
                     let (to_be_keccak_checked, assigned_sig_verif) = self.halo2_assign_sig_verify(
                         &mut ctx,
-                        &chips,
+                        &ecdsa_chip.range,
                         sign_data,
+                        sign_data_decomposed,
                         challenges,
                         &e.sig_is_valid,
                     )?;
                     assigned_sig_verifs.push(assigned_sig_verif);
                     deferred_keccak_check.push(to_be_keccak_checked);
                 }
-                // let advice_rows = ctx.advice_rows["ecdsa chip"].iter();
-                // println!(
-                //     "maximum rows used by an advice column: {}",
-                //         advice_rows.clone().max().or(Some(&0)).unwrap(),
-                // );
+                let advice_rows = ctx.advice_rows["ecdsa chip"].iter();
+                println!(
+                    "maximum rows used by an advice column: {}",
+                    advice_rows.clone().max().or(Some(&0)).unwrap(),
+                );
+                println!(
+                    "row counts: {:?}",
+                    advice_rows,
+                );
```

Comment:
the println is not suitable for product deployment.  suggest replace with log::trace! and run test with RUST_LOG=trace